Tasks/JenkinsQueueJob/jenkinsqueuejobtask.ts

```diff
@@ -22,155 +22,852 @@ var username = serverEndpointAuth['parameters']['username'];
+        for (var i in this.children) {
+            this.children[i].cut();
+        }
+    }
+
+    cut(): void {
+        this.changeState(JobState.Cut);
+        for (var i in this.children) {
+            this.children[i].cut();
+        }
+    }
+
+    getResultString(): string {
+        if (this.state == JobState.Queued) {
+            return 'Queued';
+        } else if (this.state == JobState.Done) {
+            var resultCode = this.parsedExecutionResult.result.toUpperCase();
+            // codes map to fields in http://hudson-ci.org/javadoc/hudson/model/Result.html
+            if (resultCode == 'SUCCESS') {
+                return 'Success';
+            } else if (resultCode == 'UNSTABLE') {
+                return 'Unstable';
+            } else if (resultCode == 'FAILURE') {
+                return 'Failure';
+            } else if (resultCode == 'NOT_BUILT') {
+                return 'Not built';
+            } else if (resultCode == 'ABORTED') {
+                return 'Aborted';
             } else {
-                jenkinsTaskName = parsedBody.task.name;
-                jenkinsExecutableNumber = parsedBody.executable.number;
-                jenkinsExecutableUrl = parsedBody.executable.url;
-                console.log('Jenkins job started: ' + jenkinsExecutableUrl);
-
-                if (captureConsole) {
-                    // start capturing console
-                    captureJenkinsConsole(0);
-                } else {
-                    // no console option, just create link and finish
-                    createLinkAndFinish(tl.TaskResult.Succeeded, 'Queued', 'Jenkins job successfully queued: ' + jenkinsExecutableUrl);
+                return resultCode;
+            }
+        } else return 'Unknown';
+    }
+
+    getTaskResult(finalJobState: JobState): number {
+        if (finalJobState == JobState.Queued) {
+            return tl.TaskResult.Succeeded;
+        } else if (finalJobState == JobState.Done) {
+            var resultCode = this.parsedExecutionResult.result.toUpperCase();
+            if (resultCode == "SUCCESS" || resultCode == 'UNSTABLE') {
+                return tl.TaskResult.Succeeded;
+            } else {
+                return tl.TaskResult.Failed;
+            }
+        }
+        return tl.TaskResult.Failed;
+    }
+
+    refreshJobTask(job, callback) {
+        var apiTaskUrl = addUrlSegment(this.taskUrl, "/api/json");
+        this.debug('getting job task URL:' + apiTaskUrl);
+        return request.get({ url: apiTaskUrl }, function requestCallBack(err, httpResponse, body) {
+            if (err) {
+                failError(err);
+            } else if (httpResponse.statusCode != 200) {
+                failReturnCode(httpResponse, 'Unable to retrieve job: ' + this.name);
+            } else {
+                job.parsedTaskBody = JSON.parse(body);
+                job.debug("parsedBody for: " + apiTaskUrl + ": " + JSON.stringify(job.parsedTaskBody));
+                callback();
```

Comment:
It is called actually.  I've seen connection resets come through, and these are recoverable, so I'm adding some error recovery code for all of these.